divan/chat/views.py

```diff
@@ -0,0 +1,291 @@
+        return JsonResponse(refresh, status=status.HTTP_200_OK)
+
+
+class Signout(APIView):
+    def post(self, request: HttpRequest, uid):
+        try:
+            token_process.revoke_token(uid)
+        except:
+            return Response("user id invalid ", status=status.HTTP_400_BAD_REQUEST)
+        return Response('the user is sign out', status=status.HTTP_200_OK)
+
+
+class Profile(APIView):
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@---view user profile---@@@@@@@@@@@@@@@@@@@@@@@@@@@@#
+    def get(self, request: HttpRequest, uid):
+        token_id = request.headers.get('tokenid')
+        if token_process.is_valid_token(token_id):
+            try:
+                user_obj = User.objects.get(id=uid)
+            except:
+                return Response("the user id not found", status=status.HTTP_400_BAD_REQUEST)
+            serializer = serializers.UserSerializer(user_obj)
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        else:
+            return Response("the user is not authorized", status=status.HTTP_403_FORBIDDEN)
+
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@--update user profile---@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@#
+
+    def put(self, request: HttpRequest, uid):
+        token_id = request.headers.get('tokenid')
+        if token_process.is_valid_token(token_id):
+            try:
+                user_obj = User.objects.get(id=uid)
+            except:
+                return Response("the user id not found", status=status.HTTP_400_BAD_REQUEST)
+            serializer = serializers.UserSerializer(
+                user_obj, data=request.query_params, partial=True)
+            if serializer.is_valid():
+                serializer.save()
+                return Response("ur profile is update", status=status.HTTP_200_OK)
+            else:
+                return Response("invalid data", status=status.HTTP_400_BAD_REQUEST)
+        else:
+            return Response("the user is not authorized", status=status.HTTP_403_FORBIDDEN)
+
+
+#@@@@@@@@@@@@@@@@@@@@@@@@@@@@@send friend request@@@@@@@@@@@@@@@@@@@@@@@@@@@@#
+
+class FriendRequest(APIView):
+    def post(self, request: HttpRequest, user_id, friend_id):
+        token_id = request.headers.get('tokenid')
+        if token_process.is_valid_token(token_id):
+            req = {}
+            req['user'] = user_id
+            req['friend'] = friend_id
+            serializer = serializers.FriendSerializer(data=req)
+            if serializer.is_valid():
+                serializer.save()
+                return Response("the friend request pending", status=status.HTTP_200_OK)
+            else:
+                return Response("invalid data", status=status.HTTP_400_BAD_REQUEST)
+        else:
+            return Response("the user is not authorized", status=status.HTTP_403_FORBIDDEN)
+
+
+#@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@view friend request@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@#
+
+class RequestView(APIView):
```

Comment:
Please use better names
```suggestion
class FriendRequestView(APIView):
```

Reply:
done